coalib/results/result_actions/ApplyPatchAction.py

```diff
@@ -40,14 +42,18 @@ def apply(self,
                                  pre_patch_filename + '.orig')
 
             diff = file_diff_dict[filename]
-            if diff.delete or diff.rename:
-                if isfile(pre_patch_filename):
-                    remove(pre_patch_filename)
+
             if not diff.delete:
                 new_filename = (diff.rename
                                 if diff.rename is not False
                                 else filename)
-                with open(new_filename, mode='w', encoding='utf-8') as file:
+                with open(new_filename, mode='w',
+                          encoding=detect_encoding(pre_patch_filename)) as file:
                     file.writelines(diff.modified)
 
+            if diff.delete or diff.rename:
```

Comment:
It is! If you delete the file before creating the renamed file you can't detect the encoding of the old file anymore.